model/post.go

```diff
@@ -19,31 +19,110 @@ type PostPersister interface {
 }
 
 // Post statuses
+// Draft (initial state) -> ToPublish
+//   user can update Draft to ToPublish if RecordsStatus and ImagesStatus are both Default and RecordsKey is not empty
+//   this causes server to send an Index message to Publisher
+// ToPublish -> Publishing
+//   Publisher updates status to Publishing when starting to index
+// Publishing -> PublishComplete or PublishError
+//   Publisher updates status to PublishComplete if successful, or PublishError otherwise and sets PostError to the error message
+//   having intermediate statuses for PublishComplete and PublishError means that user cannot change an Publishing status,
+//     since PublishComplete and PublishError fail UserAcceptedPostStatus()
+// PublishComplete -> Published
+//   when the server gets PublishComplete from Publisher, it sets status to Published
+// PublishError -> Error
+//   when the server gets PublishError from Publisher, it sets status to Error
+// Published -> ToUnpublish
+//   user can update Published to ToUnpublish
+//   this causes server to send an Unindex message to Publisher
+// ToUnpublish -> Unpublishing
+//   Publisher updates status to Unpublishing when starting to unindex
+// Unpublishing -> UnpublishComplete or UnpublishError
+//   Publisher updates status to to UnpublishComplete if successful, or UnpublishError otherwise and sets PostError to the error message
+//   having intermediate statuses for UnpublishComplete and UnpublishError means that user cannot change an Unpublishing status,
+//     since UnpublishComplete and UnpublishError fail UserAcceptedPostStatus()
+// UnpublishComplete -> Draft
+//   when the server gets UnpublishComplete from Publisher, it sets status to Draft
+// UnpublishError -> Error
+//   when the server gets UnpublishError from Publisher, it sets status to Error
+// Error -> ToPublish or Publishing or Unpublishing
+//   user can update Error to ToPublish if RecordsStatus and ImagesStatus are both Default
+//   Publisher can update error to Publishing or Unpublishing when retrying
+// Publisher will process the message only when status is To(Un)Publish or Error (in case the previous invocation failed)
+// Post can be deleted only in Draft or Error states and only when Records/Images statuses are in Default or Error states
+
```

Comment:
Thanks for this! Now, could you do an ASCII Art version of the diagram? (Just kidding!)